resources/src/operation/mod.rs

```diff
@@ -0,0 +1,520 @@
+    account: Option<String>,
+    funder: Option<String>,
+    starting_balance: Option<Amount>,
+    from: Option<String>,
+    to: Option<String>,
+    asset_type: Option<String>,
+    asset_code: Option<String>,
+    asset_issuer: Option<String>,
+    source_asset_type: Option<String>,
+    source_asset_code: Option<String>,
+    source_asset_issuer: Option<String>,
+    amount: Option<Amount>,
+    source_amount: Option<Amount>,
+    source_max: Option<Amount>,
+    buying_asset_type: Option<String>,
+    buying_asset_code: Option<String>,
+    buying_asset_issuer: Option<String>,
+    selling_asset_type: Option<String>,
+    selling_asset_code: Option<String>,
+    selling_asset_issuer: Option<String>,
+    offer_id: Option<i64>,
+    #[serde(rename = "price_r")] price_ratio: Option<PriceRatio>,
+    price: Option<Amount>,
+    signer_key: Option<String>,
+    signer_weight: Option<u8>,
+    master_key_weight: Option<u8>,
+    low_threshold: Option<u32>,
+    med_threshold: Option<u32>,
+    high_threshold: Option<u32>,
+    home_domain: Option<String>,
+    set_flags: Option<Vec<u32>>,
+    set_flags_s: Option<Vec<String>>,
+    clear_flags: Option<Vec<u32>>,
+    clear_flags_s: Option<Vec<String>>,
+    trustor: Option<String>,
+    trustee: Option<String>,
+    authorize: Option<bool>,
+    limit: Option<Amount>,
+    into: Option<String>,
+    name: Option<String>,
+    value: Option<String>,
+}
+
+impl<'de> Deserialize<'de> for Operation {
+    fn deserialize<D>(d: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let rep: IntermediateOperation = IntermediateOperation::deserialize(d)?;
+
+        let operation_detail: Option<OperationDetail> = match rep {
+            IntermediateOperation {
+                account: Some(account),
+                funder: Some(funder),
+                starting_balance: Some(starting_balance),
+                ..
+            } => {
+                if &rep.operation_type[..] == "create_account" {
+                    Some(OperationDetail::CreateAccount(
+                        self::create_account::CreateAccount::new(account, funder, starting_balance),
+                    ))
+                } else {
+                    None
+                }
+            }
+            IntermediateOperation {
+                from: Some(from),
+                to: Some(to),
+                asset_type: Some(asset_type),
+                amount: Some(amount),
+                source_asset_type: Some(source_asset_type),
+                source_amount: Some(source_amount),
+                source_max: Some(source_max),
+                ..
+            } => {
+                if &rep.operation_type[..] == "path_payment" {
+                    let destination_asset_identifier = AssetIdentifier::new(
+                        &asset_type,
+                        rep.asset_code,
+                        rep.asset_issuer,
+                    ).map_err(|_| {
```

Comment:
This error comes back with a string. You should just map it into the custom error instead of assuming what the error is.